scripts/create_source_tarball.sh

```diff
@@ -23,6 +23,8 @@ REPO_ROOT="$(dirname "$0")"/..
     mkdir "$SOLDIR"
     # Store the current source
     git checkout-index -a --prefix="$SOLDIR"
+    # shellcheck disable=SC2016
+    SOLDIR="$SOLDIR" git submodule foreach 'git checkout-index -a --prefix="$SOLDIR/$sm_path/"'
```

Comment:
So if I understand this correctly, it will make git copy just the code from each submodule, without the `.git` subdir? If that's the case then that interferes with your check in `CMakeLists.txt` to automatically initialize submodules in CMake. Not sure if CMake will fail or if they will get redownloaded, but either of these is not what we want and will not work with Homebrew. It must be possible to build just from the archive without getting anything from the network.

I think we need a different way to check whether a dependency is present, that does not care if it's really a submodule or just the source dumped in a subdir. Maybe we should check for presence of lib's `CMakeLists.txt` instead? Or just whether the dir exists and is non-empty?

Reply:
That's true! Initially I had implemented it to do just determine the need for submodule initialisation by finding out whether the directory was just empty. I think I will just implement it like this again, because that will make the resulting tarball usable again.